### ISSUE TITLE
Could not find package when using composer repository that provide lazy providers

### ISSUE THREAD

**Issue by IndraGunawan** [composer/composer#6993]
How to reproduce

- use composer repositry that provide lazy providers like `packagist.com` or `pakethub.com` and add the repository config to `composer.json`
- add `symfony/symfony` package to the composer repository provider. so when you access `packages.json` from the composer repository hosting you will get 
```json
{
"packages":{"symfony/symfony":{...}},
"search":"...",
"providers-lazy-url":".../p/%package%.json",
"mirrors":[{...}]
}
```
- run `composer require symfony/security-bundle -vvv` at project directory.

Here the response
```
[InvalidArgumentException]
  Could not find package symfony/security-bundle.
```

**Comment by Seldaek** [comment]
You need to make sure to add symfony/security-bundle to the repository as well if I understand your workflow correctly. With automatic mirroring on packagist.com I believe it should do it for you though.

**Comment by IndraGunawan** [comment]
i tried with packagist.com and it doesn't work. i need to add manually the symfony/security-package from the web ui

**Comment by Seldaek** [comment]
Ah well sorry I had forgotten about this but indeed this is an issue which we can only fix in Composer 2.0 though so in the meantime you should indeed add replaced packages by hand.

**Comment by naderman** [comment]
The alternative in the case of symfony/symfony on packagist.com is to delete the mirrored symfony/symfony package, which will be automatically added back by this process, but it'll also mirror symfony/security-bundle.

**Comment by Seldaek** [comment]
Fixed in 2.0 so closing
